modules/core/keeper/msg_server.go

```diff
@@ -764,7 +764,7 @@ func (k Keeper) ChannelUpgradeTry(goCtx context.Context, msg *channeltypes.MsgCh
 	if err != nil {
 		ctx.Logger().Error("channel upgrade try failed", "error", errorsmod.Wrap(err, "channel upgrade try failed"))
 		if channeltypes.IsUpgradeError(err) {
-			_ = k.ChannelKeeper.WriteErrorReceipt(ctx, msg.PortId, msg.ChannelId, upgrade.Fields, err.(*channeltypes.UpgradeError))
+			_ = k.ChannelKeeper.WriteErrorReceipt(ctx, msg.PortId, msg.ChannelId, err.(*channeltypes.UpgradeError))
```

Comment:
as a general note here, if we used direct type checking instead of `IsUpgradeError` then we don't need to do this type conversion when passing the arg, right?

Reply:
correct, I don't feel super strongly about which way we go with this, but I think we can do this in a follow up if we want to!